### PR TITLE
feat: DAOs basic support

### PR DESCRIPTION
**Pull request by YoLoADR**
https://user-images.githubusercontent.com/16499678/203776115-931cde06-7eda-4c11-b5ea-0c3ee9a03d62.mov

### CODE REVIEW

**Comment by zxxma** [comment]
Need an update on this @gmspacex to avoid too much conflict

File: .env [hunk 1]
```diff
@@ -27,3 +33,13 @@ PRICES_SERVICE_URI=prices.mainnet.teritori.com:443
...
 SELECTABLE_NETWORKS_IDS=teritori,teritori-testnet,ethereum,ethereum-goerli
+
+DAO_CW20_CODE_ID=99
+DAO_FACTORY_CODE_ID=100
+DAO_CORE=101
+DAO_PRE_PROPOSE_SINGLE=102
+DAO_PROPOSAL_SINGLE=103
+DAO_VOTING_CW20_STAKED=104
+DAO_CW20_STAKE=105
+TERITORI_DAO_FACTORY_CONTRACT_ADDRESS=tori1r29chp8ufwgx9u3wr4sfk050aardhkzwve7nht6y06gvlaqutr3qt83z6r
```
Comment (n0izn0iz) [inline .env:44]: all these vars should come from network config

File: go/internal/indexerdb/dao.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package indexerdb
+
+type Dao struct {
+	Id                     uint32 `gorm:"primaryKey;autoIncrement"`
```
Comment (n0izn0iz) [inline go/internal/indexerdb/dao.go:4]: this id should not be managed by the indexer, it should be derived from onchain data uniquely identifying this dao, like main contract address

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: go/cmd/teritori-dapp-backend/main.go [hunk 3]
```diff
@@ -86,6 +88,12 @@ func main() {
 	if err != nil {
 		panic(errors.Wrap(err, "failed to access db"))
 	}
+
+	err = indexerdb.MigrateDB(indexerDB)
+	if err != nil {
+		panic(errors.Wrap(err, "failed migrate database models"))
+	}
```
[the remark this answers is not in the document]
Comment (3mp8r3) [reply go/cmd/teritori-dapp-backend/main.go]: done

File: go/internal/indexerdb/dao.go (hunk 2 above)
Comment (3mp8r3) [reply go/internal/indexerdb/dao.go]: done

File: .env (hunk 1 above)
Comment (3mp8r3) [reply .env]: k

**Comment by zxxma** [comment]
gg wp!

**Upcoming needed fix:** 

- [ ] - Auto-add ".tori" at the end of the input
- [ ] - Allow to add with TNS in "Add member" inputs
- [ ] -  Add "voting power" above the input (Add Member screen)
- [ ] - Make UPP content at least as high as the page (like UPP NFTs tab )
- [ ] - In /orgs list screen, filter by creation date
- [ ] - Close modal after member invitation

**Pull request merged**